java/vector/src/test/java/org/apache/arrow/vector/util/DecimalUtilityTest.java

```diff
@@ -41,7 +41,8 @@ public void testSetLongInDecimalArrowBuf() {
       ) {
         int [] intValues = new int [] {Integer.MAX_VALUE, Integer.MIN_VALUE, 0};
         for (int val : intValues) {
-          buf.clear();
```

Comment:
given these changes, I think maybe we should keep clear() and remove all the rest.  adding more calls for readerIndex and writeIndex seems like it is moving in the wrong direction.

Reply:
sounds good. I have revised the PR to preserve the method.